Batch8CucumberSeleniumAPI/src/test/java/stepdefs/ui/SearchStepDefs.java

```diff
@@ -116,6 +117,32 @@ public void i_am_able_to_see_product_description_and_detail_in_new_tab() {
 		testContextUI.getProductDescriptionObjects().switchToDefaultWindowTab();	
 	}
 	
+	//Product Refinements
+	@When("I search for product as a {string}")
+	public void i_search_for_product_as_a(String productName) {
+		testContextUI.getCmnPageObjects().SetSearchTextBox(productName);
+		testContextUI.getCmnPageObjects().ClickOnSearchButton();
+	}
+
+	@Then("Below Search product refinements should displayed")
+	public void below_Search_product_refinements_should_displayed(List<String> refinementlist ) {
+
+		for(int i=0;i<refinementlist.size();i++)
+		{
+			try
+			{
+               testContextUI.getSearchPageObjects().validateProductRefinementList(refinementlist.get(i));
+			}
+			catch(Exception e)
+			{
+				scn.write("Exception thrown: "+e.getMessage());
+			}
+		}
+		
+		testContextUI.getSearchPageObjects().DisplayedProductRefinementList();
```

Comment:
why do have these validation methods.
is this method call not enough to validate.:
testContextUI.getSearchPageObjects().validateProductRefinementList(refinementlist.get(i));?